src/LanguageServer/Protocol.TestUtilities/LanguageServer/AbstractLanguageServerProtocolTests.cs

```diff
@@ -698,6 +709,33 @@ public async Task OpenDocumentAsync(Uri documentUri, string? text = null, string
                 await ExecuteRequestAsync<LSP.DidOpenTextDocumentParams, object>(LSP.Methods.TextDocumentDidOpenName, didOpenParams, CancellationToken.None);
             }
 
+            /// <summary>
+            /// Opens a document in the workspace only, and waits for workspace operations.
+            /// Use <see cref="OpenDocumentAsync(Uri, string, string)"/> if the document should be opened in LSP"/>
+            /// </summary>
+            public async Task OpenDocumentInWorkspaceAsync(DocumentId documentId, bool openAllLinkedDocuments, SourceText? text = null)
```

Comment:
previously, some tests opened text buffers to simulate opening a file in the workspace (and setting things like active document context).

We don't actually need the buffers - instead we can just open the documents in the workspace directly.